pdm/project/core.py

```diff
@@ -515,3 +516,25 @@ def make_candidate_info_cache(self) -> CandidateInfoCache:
 
     def make_hash_cache(self) -> HashCache:
         return HashCache(directory=self.cache("hashes").as_posix())
+
+    def find_interpreters(self, python_spec: str) -> Iterable[str]:
+        """Return an iterable of interpreter paths that matches the given specifier,
+        which can be:
+            1. a version specifier like 3.7
+            2. an absolute path
+            3. a short name like python3
+        """
+        import pythonfinder
+
+        if python_spec and not all(c.isdigit() for c in python_spec.split(".")):
+            if Path(python_spec).exists():
+                python_path = find_python_in_path(python_spec)
+                if python_path:
+                    yield python_path
+                return
+            args = [python_spec]
+        else:
+            args = [int(v) for v in python_spec.split(".") if v != ""]
+        finder = pythonfinder.Finder()
+        for entry in finder.find_all_python_versions(*args):
+            yield entry.path.as_posix()
```

Comment:
are you getting much benefit from using a generator here?

seems to me all the lists in the method are allocated to memory, so returning each item one by one as an iterator doesn't offer any performance benefits.

It *would* if the upstream method `finder.find_all_python_versions` returned an iterator, but it doesn't- it returns an allocated list.

Reply:
Not much, just provides a more neat interface for subclass methods:

```python
def find_interpreters(self, python_spec: str) -> Iterable[str]:
    yield some
    yield another
    yield from super().find_interpreters(python_spec)
```
Instead of:
```python
def find_interpreters(self, python_spec: str) -> Iterable[str]:
    results = []
    results.append(some)
    results.append(another)
    results.extend(super().find_interpreters(python_spec))
    return results
```